buildSrc/src/main/groovy/org/apache/beam/gradle/kafka/KafkaTestUtilities.groovy

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * License); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.gradle.kafka
+
+import org.gradle.api.Project
+import org.gradle.api.artifacts.ConfigurationContainer
+import org.gradle.api.tasks.testing.Test
+
+import javax.inject.Inject
+
+class KafkaTestUtilities {
+  abstract static class KafkaBatchIT extends Test {
+
+    @Inject
+    KafkaBatchIT(String delimited, String undelimited, Boolean sdfCompatible, ConfigurationContainer configurations, Project runningProject){
+      group = "Verification"
+      description = "Runs KafkaIO IT tests with Kafka clients API $delimited"
+      outputs.upToDateWhen { false }
+      testClassesDirs = runningProject.findProject(":sdks:java:io:kafka").sourceSets.test.output.classesDirs
+      classpath =  configurations."kafkaVersion$undelimited" + runningProject.sourceSets.test.runtimeClasspath + runningProject.findProject(":sdks:java:io:kafka").sourceSets.test.runtimeClasspath
+
+      def pipelineOptions = [
+        '--sourceOptions={' +
+        '"numRecords": "1000",' +
+        '"keySizeBytes": "10",' +
+        '"valueSizeBytes": "90"' +
+        '}',
+        "--readTimeout=120",
+        "--kafkaTopic=beam",
+        "--withTestcontainers=true",
+        "--kafkaContainerVersion=5.5.2",
```

Comment:
Could we have these as variables that are then concatenated in the pipelineOptions?